frame/examples/kitchensink/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+/// The kitchen-sink catalog of the the FRAME macros and their various syntax options.
+///
+/// This example does not focus on pallet instancing, `dev_mode`, and does nto include any 'where'
+/// clauses on `T`. These will both incur additional complexity to the syntax, but are not discussed
+/// here.
+// TODO: link to a particular example pallet that highlights each of these.
```

Comment:
probably not needed. 

But what about a parent crate at `frame/example` that just links to each example and shows what they do? it would contain no code.

Reply:
Added